packages/desktop-client/src/components/common/Link.tsx

```diff
@@ -136,28 +135,34 @@ const InternalLink = ({
 };
 
 type LinkProps =
-  | ({
-      variant: 'button';
-    } & ButtonLinkProps)
-  | ({ variant?: 'internal' } & InternalLinkProps)
-  | ({ variant?: 'external' } & ExternalLinkProps)
-  | ({ variant?: 'text' } & TextLinkProps);
-
-export function Link({ variant = 'internal', ...props }: LinkProps) {
-  switch (variant) {
-    case 'internal':
-      return <InternalLink {...props} />;
-
-    case 'external':
-      return <ExternalLink {...props} />;
-
-    case 'button':
-      return <ButtonLink {...props} />;
-
-    case 'text':
-      return <TextLink {...props} />;
+  | ({ variant: 'button' } & ButtonLinkProps)
+  | ({ variant: 'internal' } & InternalLinkProps)
+  | ({ variant: 'external' } & ExternalLinkProps)
+  | ({ variant: 'text' } & TextLinkProps);
+
+export function Link(props: LinkProps) {
+  switch (props.variant) {
+    case 'internal': {
+      const { variant: _, ...internalProps } = props;
+      return <InternalLink {...internalProps} />;
+    }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏a small simplification we could do here
```suggestion
export function Link({ variant, ...props }: LinkProps) {
  switch (variant) {
    case 'internal':
      return <InternalLink {...props} />;
```

Reply:
Other than this comment (and the ones by matt)  - LGTM!

Reply:
Tried this initially but compiler is complaining:
![image](https://github.com/user-attachments/assets/7636ce4d-6108-4e48-951f-381984fbda4c)

Reply:
ah.. weird.